starfish/intensity_table/intensity_table.py

```diff
@@ -384,6 +386,54 @@ def to_features_dataframe(self) -> pd.DataFrame:
         """
         return pd.DataFrame(dict(self[Features.AXIS].coords))
 
+    def to_expression_matrix(self, regions: Optional[regional.many]=None) -> ExpressionMatrix:
+        """Generates a cell x gene count matrix where each cell is annotated with spatial metadata
+
+        Parameters
+        ----------
+        regions: Optional[regional.Many]
+            cell segmentation results that were used to assign points to cells. If not provided, the
+            centers of the cells will be estimated by taking the midpoint between the extreme-valued
+            spots on each axis.
+
+        Returns
+        -------
+        ExpressionMatrix :
+            cell x gene expression table
+        """
+
+        # create the 2-d counts matrix
+        grouped = self.to_features_dataframe().groupby(['cell_id', 'target'])
+        counts = grouped.count().iloc[:, 0].unstack().fillna(0)
+
+        if regions:
+            # counts.index stores cell_id, extract cell information from the regional.many object
+            metadata = {
+                "area": ("cells", [regions[id_].area for id_ in counts.index]),
+                "x": ("cells", [regions[id_].center[0] for id_ in counts.index]),
+                "y": ("cells", [regions[id_].center[1] for id_ in counts.index]),
+                "z": ("cells", np.zeros(counts.shape[0]))
+            }
+        else:
+            grouped = self.to_features_dataframe().groupby(['cell_id'])[['x', 'y', 'z']]
+            min_ = grouped.min()
+            max_ = grouped.max()
+            coordinate_df = min_ + (max_ - min_) / 2
```

Comment:
I don't think you need to do this trick as python doesn't really have a number range (it automatically promotes types).

Reply:
I don't understand. What I'm attempting to capture here is the midpoint between extreme values of spots associated with cells, to build a "center" for each cell.

Reply:
so isn't that just (min_ + max_) / 2?

The min_ + (max_ - min_) / 2 trick is one that people do to prevent overflow.